### PR TITLE
Added content-type to SetRequest and blob storage to test

### PR DESCRIPTION
**Pull request by jigargandhi**
# Description

Added Content-Type to SetRequest.

## Issue reference
https://github.com/dapr/dapr/issues/2892

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by jigargandhi** [comment]
How this will work:

I have added Content Type to common.proto in dapr/dapr and also set the value in grpc SetRequest call see branch: https://github.com/jigargandhi/dapr/tree/fix-proto-issues
With this change if content type is present as a part of SetRequest struct it will be used.

Once this PR is merged, then I will update the components contrib in dapr and create a PR for that branch.

**Comment by CodeMonkeyLeet** [comment]
@artursouza this brings back the changes from reverted #1143, are we ready to take the changes now?

File: state/requests.go [hunk 1]
```diff
@@ -45,11 +45,12 @@ type DeleteStateOption struct {
...
-	Value    interface{}       `json:"value"`
-	ETag     *string           `json:"etag,omitempty"`
-	Metadata map[string]string `json:"metadata,omitempty"`
-	Options  SetStateOption    `json:"options,omitempty"`
+	Key         string            `json:"key"`
+	Value       interface{}       `json:"value"`
+	ETag        *string           `json:"etag,omitempty"`
+	Metadata    map[string]string `json:"metadata,omitempty"`
+	Options     SetStateOption    `json:"options,omitempty"`
+	ContentType string            `json:"contentType,omitempty"`
```
Comment (artursouza) [inline state/requests.go:53]: Please, use metadata for now. We need to discuss ContentType for all APIs and not only this one.

**Review by artursouza — CHANGES_REQUESTED** [review]

**Comment by beiwei30** [comment]
@jigargandhi pls. check Artur's review comments above (https://github.com/dapr/components-contrib/pull/1299#discussion_r753594798)

File: state/requests.go (hunk 1 above)
Comment (jigargandhi) [reply state/requests.go:53]: Hi @artursouza will take this up in state-management channel in discord.

**Comment by jigargandhi** [comment]
Closing this PR and creating a new one with signoff

**Pull request closed without merging**